sdk-core/transactions-factories/token_management_transactions_factory.md

```diff
@@ -0,0 +1,174 @@
+        gasPrice?: IGasPrice;
+        gasLimit?: IGasLimit;
+        guardian? : IAddress;
+    })
+
+    createIssueNonFungibleTransaction({
+        sender: IAddress;
+        tokenName: string;
+        tokenTicker: string;
+        canFreeze: boolean;
+        canWipe: boolean;
+        canPause: boolean;
+        canTransferNFTCreateRole: boolean;
+        canChangeOwner: boolean;
+        canUpgrade: boolean;
+        canAddSpecialRoles: boolean;
+
+        // Optionals:
+        transactionNonce?: INonce;
+        value?: ITransactionValue;
+        gasPrice?: IGasPrice;
+        gasLimit?: IGasLimit;
+        guardian? : IAddress;
+    }): Transaction;
+
+    createRegisterMetaESDTTransaction({
+        sender: IAddress;
+        tokenName: string;
+        tokenTicker: string;
+        numDecimals: number;
+        canFreeze: boolean;
+        canWipe: boolean;
+        canPause: boolean;
+        canTransferNFTCreateRole: boolean;
+        canChangeOwner: boolean;
+        canUpgrade: boolean;
+        canAddSpecialRoles: boolean;
+
+        // Optionals:
+        transactionNonce?: INonce;
+        value?: ITransactionValue;
+        gasPrice?: IGasPrice;
+        gasLimit?: IGasLimit;
+        guardian? : IAddress;
+    }): Transaction;
+
+    createRegisterAndSetAllRolesTransaction({
+        sender: IAddress;
+        tokenName: string;
+        tokenTicker: string;
+        tokenType: RegisterAndSetAllRolesTokenType;
+        numDecimals: number;
+
+        // Optionals:
+        transactionNonce?: INonce;
+        value?: ITransactionValue;
+        gasPrice?: IGasPrice;
+        gasLimit?: IGasLimit;
+        guardian? : IAddress;
+    }): Transaction;
+
+    createSetBurnRoleGloballyTransaction({
+        sender: IAddress;
+        tokenIdentifier: string;
+
+        // Optionals:
+        transactionNonce?: INonce;
+        value?: ITransactionValue;
+        gasPrice?: IGasPrice;
+        gasLimit?: IGasLimit;
+        guardian? : IAddress;
+    }): Transaction;
+
+    createUnsetBurnRoleGloballyTransaction({
+        sender: IAddress;
+        tokenIdentifier: string;
+
+        // Optionals:
+        transactionNonce?: INonce;
+        value?: ITransactionValue;
+        gasPrice?: IGasPrice;
+        gasLimit?: IGasLimit;
+        guardian? : IAddress;
+    }): Transaction;
+
+    createSetSpecialRoleOnFungibleTransaction({
+        sender: IAddress;
+        user: IAddress;
+        tokenIdentifier: string;
+        addRoleLocalMint: boolean;
+        addRoleLocalBurn: boolean;
+
+        // Optionals:
+        transactionNonce?: INonce;
+        value?: ITransactionValue;
+        gasPrice?: IGasPrice;
+        gasLimit?: IGasLimit;
+        guardian? : IAddress;
+    }): Transaction;
+
+    createSetSpecialRoleOnSemiFungibleTransaction({
+        
+    })
+```
+
+If the language supports it, the input arguments objects may be designed using interfaces and inheritance. For example, in TypeScript:
```

Comment:
plain DTOs will suffice, I guess, no need to overcomplicate with interfaces & only one instance

Reply:
True for Go :+1: 

For JavaScript, developers can provide objects constructed in an ad-hoc fashion. For example:

https://github.com/multiversx/mx-sdk-js-core/blob/main/src/tokenOperations/tokenOperationsFactory.test.net.spec.ts#L131

Reply:
Removed this detail from the specs.